extensions/standard-processors/processors/GetFile.cpp

```diff
@@ -228,42 +228,49 @@ void GetFile::pollListing(std::queue<std::string> &list, const GetFileRequest &r
 bool GetFile::acceptFile(std::string fullName, std::string name, const GetFileRequest &request) {
   logger_->log_trace("Checking file: %s", fullName);
 
+#ifdef WIN32
+  struct _stat64 statbuf;
+  if (_stat64(fullName.c_str(), &statbuf) != 0) {
+    return false;
+  }
+#else
   struct stat statbuf;
+  if (stat(fullName.c_str(), &statbuf) != 0) {
+    return false;
+  }
+#endif
+  uint64_t file_size = gsl::narrow<uint64_t>(statbuf.st_size);
+  uint64_t modifiedTime = gsl::narrow<uint64_t>(statbuf.st_mtime) * 1000;
```

Comment:
Is this safe (both for the narrowing before scaling and the type usage)? Shouldn't we use `std::difftime`?

Reply:
you are right about our inconsistent time representations (we use `uint64_t`, `int64_t` and `time_t` in `TimeUtil.h` to represent time) we should unify it, based on the POSIX and Windows documentations, as I understand we may (cautiously) assume that `st_mtime` yields the Unix time, so these narrowings and the subsequent scaling should be fine